src/main/webapp/careerquiz.js

```diff
@@ -0,0 +1,76 @@
+/*
+* Purpose: recieves HTTP promise response from Career Quiz servlet, displays questions and responses 
+*/
+
+function getQuestionsAndChoices() {
+ const responsePromise = fetch('/careerquiz');
+ responsePromise.then(handleResponse);
+}
+
+function handleResponse(response) {
+ const responsePromise = response.json();
+ responsePromise.then(addToDom);
+}
+
+/* Adds authors and comments to html page, displaying author: comment.*/
+function addToDom(questionsAndChoicesList) {
+ const questionsDiv = document.getElementById('questions-and-choices-div');
+ questionsDiv.innerHTML = '';
+ for (let i = 0; i < questionsAndChoicesList.length; i++){
+ question = questionsAndChoicesList[i].question;
+ choices = questionsAndChoicesList[i].choices
+ questionsDiv.appendChild(
+ createQuestionAndChoices(question, choices));
+ }
+}
+
+function createQuestionAndChoices(question, choices) {
+ const formContainer = document.createElement('div');
```

Comment:
Since its constant, would the type be uppercase?

Reply:
The style guide for JS here doesn't always have const variables as uppercase: https://google.github.io/styleguide/jsguide.html. I interpreted it as literal constants should be uppercase, like hardcoded strings or integers? I could be wrong though.